src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java

```diff
@@ -198,10 +228,23 @@ else if (len < 2000)
 
         if (ENABLE_BOOKKEEPING)
         {
-            logger.info("Thread " + threadId() + " returned array "
-                    + System.identityHashCode(buf));
             Integer arrayId = System.identityHashCode(buf);
-            if (bookkeeping.remove(arrayId) == null)
+            logger.info("Thread " + threadId() + " returned array "
+                    + arrayId);
+            if (bookkeeping.containsKey(arrayId))
+            {
+                returnedBookkeeping.put(arrayId, new ReturnedBufferBookkeepingInfo(bookkeeping.get(arrayId), getStackTrace()));
+                bookkeeping.remove(arrayId);
+            }
+            else if (returnedBookkeeping.containsKey(arrayId))
+            {
+                logger.info("Thread " + threadId()
+                    + " returned a previously-returned buffer at\n"
+                    + getStackTraceAsString() +
+                    "previously returned at\n" +
+                    stackTraceAsString(returnedBookkeeping.get(arrayId).deallocTrace));
+            }
+            else
             {
                 logger.info("Thread " + threadId()
```

Comment:
As I'm doing debugging, I'm finding I wish we had the size of the buffer as well here.